pkgs/applications/science/misc/gephi/default.nix

```diff
@@ -1,37 +1,64 @@
-{stdenv, fetchurl, jdk}:
+{ stdenv, fetchFromGitHub, callPackage, jdk, maven, javaPackages }:
 
-with stdenv.lib;
+let
+  version = "0.9.2";
 
-let version = "0.9.2"; in
-stdenv.mkDerivation {
-  name = "gephi-${version}";
-
-  src = fetchurl {
-    url = "https://github.com/gephi/gephi/releases/download/v${version}/gephi-${version}-linux.tar.gz";
-    sha256 = "1wr3rka8j2y10nnwbg27iaxkbrp4d7d07ccs9n94yqv6wqawj5z8";
+  src = fetchFromGitHub {
+    owner = "gephi";
+    repo = "gephi";
+    rev = "v${version}";
+    sha256 = "0kqp2nvnsb55j1axb6hk0mlw5alyaiyb70z0mdybhpqqxyw2da2r";
   };
 
-  meta = {
-    inherit version;
-    description = "A platform for visualizing and manipulating large graphs";
-    homepage = https://gephi.org;
-    license = licenses.gpl3;
-    maintainers = [maintainers.taeer];
-    platforms = platforms.linux;
+  # perform fake build to make a fixed-output derivation out of the files downloaded from maven central (120MB)
+  deps = stdenv.mkDerivation {
+    name = "gephi-${version}-deps";
+    inherit src;
+    buildInputs = [ jdk maven ];
+    buildPhase = ''
+      while mvn package -Dmaven.repo.local=$out/.m2 -Dmaven.wagon.rto=5000; [ $? = 1 ]; do
+        echo "timeout, restart maven to continue downloading"
+      done
+    '';
+    # keep only *.{pom,jar,sha1,nbm} and delete all ephemeral files with lastModified timestamps inside
+    installPhase = ''find $out/.m2 -type f -regex '.+\(\.lastUpdated\|resolver-status\.properties\|_remote\.repositories\)' -delete'';
```

Comment:
Could this be in future a generic fetcher for maven modules?

Reply:
maybe.
but if would have a lot of configurable params.
It is the 3rd place where the code is copy-pasted, and all 3 have little differences.
I am not sure if taking the common part out to a generic function would reduce the number of code lines

Reply:
Another reason to postpone the generic builders for ```maven``` and ```gradle```: in ```nixpkgs``` these are very few java programs built from sources, usually download and unpack. As I add them (in PR and privately for non-opensource programs), it reveals problems in how java is handled in ```nixpkgs``` (for example https://github.com/NixOS/nixpkgs/issues/37364). If we will make genetic builders and then refactor ```openjdk```, it might result in refactoring those genetic builders too. So far the builders are ad-hoc, it is simpler to make changes in jdk derivations without being afraid to break the builders.

Reply:
Yeah, so far packaging java applications was pretty straight-forward and building from source did not provide significant advantages.